src/dupradar/dupradar.r

```diff
@@ -0,0 +1,191 @@
+output_duprate_exp_denscurve_mqc <- ifelse(is.null(par$output_duprate_exp_denscurve_mqc), paste0(output_prefix, "_duprateExpDensCurve_mqc.txt", sep=""), par$output_duprate_exp_denscurve_mqc)
+output_expression_histogram <- ifelse(is.null(par$output_expression_histogram), paste0(output_prefix, "_expressionHist.pdf", sep=""), par$output_expression_histogram)
+output_intercept_slope <- ifelse(is.null(par$output_intercept_slope), paste0(output_prefix, "_intercept_slope.txt", sep=""), par$output_intercept_slope)
+
+bamRegex <- "(.+)\\.bam$"
+
+
+if(!(grepl(bamRegex, input_bam) && file.exists(input_bam) && (!file.info(input_bam)$isdir))) stop("First argument '<input.bam>' must be an existing file (not a directory) with '.bam' extension...")
+if(!(file.exists(annotation_gtf) &&  (!file.info(annotation_gtf)$isdir))) stop("Third argument '<annotation.gtf>' must be an existing file (and not a directory)...")
+if(is.na(stranded) || (!(stranded %in% (0:2)))) stop("Fourth argument <strandDirection> must be a numeric value in 0(unstranded)/1(forward)/2(reverse)...")
+if(is.na(threads) || (threads<=0)) stop("Fifth argument <nbThreads> must be a strictly positive numeric value...")
+
+# Debug messages (stderr)
+message("Input bam      (Arg 1): ", input_bam)
+message("Output basename(Arg 2): ", output_prefix)
+message("Input gtf      (Arg 3): ", annotation_gtf)
+message("Strandness     (Arg 4): ", c("unstranded", "forward", "reverse")[stranded])
+message("paired_end     (Arg 5): ", paired_end)
+message("Nb threads     (Arg 6): ", threads)
+message("R package loc. (Arg 7): ", ifelse(length(args) > 4, args[5], "Not specified"))
+
+
+# Load / install packages
+if (length(args) > 5) { .libPaths( c( args[6], .libPaths() ) ) }
+if (!require("dupRadar")) {
+  if (!requireNamespace("BiocManager", quietly = TRUE)) {
+    install.packages("BiocManager")
+  }
+  BiocManager::install("dupRadar", update = TRUE, ask=FALSE)
+  library("dupRadar")
+}
+if (!require("parallel")) {
+    library("parallel")
+}
+
+
+# Duplicate stats
+dm <- analyzeDuprates(input_bam, annotation_gtf, stranded, paired_end, threads)
+print("analyzeDuprates done")
+write.table(dm, file=output_dupmatrix, quote=F, row.name=F, sep="\t")
+print("write.table done")
+
+# 2D density scatter plot
+pdf(output_duprate_exp_densplot)
+print("pdf done")
+duprateExpDensPlot(DupMat=dm)
+title("Density scatter plot")
+mtext(output_prefix, side=3)
+dev.off()
+print("duprateExpDensPlot done")
+fit <- duprateExpFit(DupMat=dm)
+print("duprateExpFit done")
+cat(
+    paste("- dupRadar Int (duprate at low read counts):", fit$intercept),
+    paste("- dupRadar Sl (progression of the duplication rate):", fit$slope),
+    fill=TRUE, labels=output_prefix,
+    file=output_intercept_slope, append=FALSE
+)
+
+# Create a multiqc file dupInt
+sample_name <- gsub("Aligned.sortedByCoord.out.markDups", "", output_prefix)
+line="#id: DupInt
```

Comment:
Always use `<-` for assignment

```suggestion
line <- "#id: DupInt
```